spring-integration-core/src/main/resources/META-INF/spring.integration.default.properties

```diff
@@ -4,3 +4,4 @@ spring.integration.channels.maxBroadcastSubscribers=0x7fffffff
 spring.integration.taskScheduler.poolSize=10
 spring.integration.messagingTemplate.throwExceptionOnLateReply=false
 spring.integration.messagingAnnotations.require.componentAnnotation=false
+spring.integration.messagingGateway.convertReceiveMessage=false
```

Comment:
There's a __big__ difference between "relying on a bug" and "working around a bug". This will break any such work-arounds. 

We should make life as painless as possible for someone who has had to already spend time and effort to find a work-around to a problem - (s)he may even have thought this is the way it's supposed to work and did not even consider it a workaround.

I therefore __strongly__ believe we should set it to `true` by default if we decide to have this flag.

Here's another solution that keeps the current behavior without the need for the property...

```java
if (shouldReturnMessage) {
    if (result.getPayload() instanceof Message) {
        return result.getPayload(); // unwrap for backwards compatibility
    else {
        return result;
    }
}

Perhaps log a warning (once) if we unwrap...

"In 5.0 the behavior for Message<?> return types will be such that a Message<?> wrapped in a Message<?> will not be unwrapped".